### PR TITLE
Check build stability

### PR DESCRIPTION
**Pull request by timkpaine**
Opening because it seems the build went unstable between when I opened the last PR and its CI ran, and when I merged it. Will debug on this PR.

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mattip** [comment]
Upstream excludes `grpcio==1.56.0` for darwin. We may need that exclusion after #123

**Comment by timkpaine** [comment]
I still see the `!=` pin here: https://github.com/ray-project/ray/blob/ray-2.8.0/python/setup.py#L264

**Comment by mattip** [comment]
I am not sure what is going on.

Half of the builds [are failing](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=828622&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=6d4b912b-175d-51da-0fd9-4d30fe1eb4e7&l=110494) to create a virtural environment with
```
KeyError: Dist(channel='conda-forge', dist_name='wcwidth-0.2.12-pyhd8ed1ab_0', \
   name='wcwidth', fmt='.conda', version='0.2.12', build_string='pyhd8ed1ab_0', \
   build_number=0, base_url=None, platform=None)
```

It seems wcwidth was updated recently. Perhaps the conda locking mechanism is getting confused here?

I will restart the failing jobs.

**Comment by mattip** [comment]
There should be a way of ensuring `conda install` gets a consistent state, even when packages are updating. @isuruf is this something you have seen elsewhere, where a CI job fails because of new packages?

**Comment by mattip** [comment]
[here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=828253&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=6d4b912b-175d-51da-0fd9-4d30fe1eb4e7&l=110417) is an example of the failure from a merge.

**Comment by timkpaine** [comment]
Yeah it's another key error, I usually see this with mamba. I wonder if it's a general conda bug that just happens to coincide with our merges

**Comment by timkpaine** [comment]
https://github.com/conda-forge/conda-smithy/issues/1805

**Comment by timkpaine** [comment]
Maybe @conda-forge/core knows more

**Comment by jaimergp** [comment]
Sorry about this! It's essentially https://github.com/conda/conda-libmamba-solver/issues/386. We are tracking down the bug and I think Marcel found the root cause in https://github.com/conda/conda-libmamba-solver/issues/393. Reverting temporarily to `conda_built_tool: mambabuild` in conda-forge.yml should fix it.

**Comment by jaimergp** [comment]
@conda-forge-admin, please rerender

**Comment by jaimergp** [comment]
@conda-forge-admin, please rerender

File: conda-forge.yml [hunk 1]
```diff
@@ -4,3 +4,4 @@ github:
   tooling_branch_name: main
 conda_build:
   pkg_format: '2'
+conda_build_tool: mambabuild
```
Comment (mbargull) [inline conda-forge.yml:7]: We have a new `conda-smithy` release that has this a the default for now.

Comment (mattip) [reply conda-forge.yml:7]: Thanks for tracking this down

**Comment by jaimergp** [comment]
> Yeah it's another key error, I usually see this with mamba.

Btw @timkpaine, by "mamba" there, do you mean `mambabuild` or `conda-libmamba-solver`? AFAIK this KeyError issue is a conda-libmamba-solver only thing, but maybe you had seen it with mambabuild too? Just double checking while we figure this out.

**Comment by timkpaine** [comment]
I have frequently seen this when installing mamba and boa and then trying to render a recipe with `conda mambabuild`, for ~4-5 months now. I had assumed it was due to incompatibilities between released versions of mamba, Conda-build, and boa (and boa in particular has been due for a new release for quite some time).

**Comment by timkpaine** [comment]
@conda-forge-admin please rerender

**Comment by mbargull** [comment]
> I have frequently seen this when installing mamba and boa and then trying to render a recipe with `conda mambabuild`, for ~4-5 months now. I had assumed it was due to incompatibilities between released versions of mamba, Conda-build, and boa (and boa in particular has been due for a new release for quite some time).

Interesting. If you happen to stumble on such a case again, please let us know (with a`conda info`/`conda list` outputs and recipe, ideally)!

**Comment by isuruf** [comment]
The KeyError used to happen (not frequently) and went away after some time. For eg: https://github.com/conda-forge/rust-feedstock/pull/75#issuecomment-783375017

**Pull request merged**